pkg/status/status_collector.go

```diff
@@ -274,6 +274,26 @@ func (d *statusCollector) getCNIChainingStatus() *models.CNIChainingStatus {
 	}
 }
 
+func (d *statusCollector) getBGPStatus(ctx context.Context) *models.BGPStatus {
+	healthy, msg := d.statusParams.BGPStatusGetter.GetBGPPeerStatus(ctx)
+	var mode string
+	if !d.statusParams.DaemonConfig.EnableBGPControlPlane {
+		return &models.BGPStatus{
+			Mode: models.BGPStatusStateDisabled,
+			Msg:  "BGP control plane is disabled",
+		}
+	}
```

Comment:
nit: Let's do this check before we call `GetBGPPeerStatus()`, since we know that it will not return anything meaningful if BGP is disabled